var/spack/repos/builtin/packages/relion/package.py

```diff
@@ -34,29 +35,55 @@ class Relion(CMakePackage, CudaPackage):
     homepage = "http://http://www2.mrc-lmb.cam.ac.uk/relion"
     git      = "https://github.com/3dem/relion.git"
 
-    version('develop')
-    version('2.1', tag='2.1')
-    version('2.0.3', tag='2.0.3')
+    version('3.0_beta', git='https://bitbucket.org/scheres/relion-3.0_beta.git')
+    version('2.1', git='https://github.com/3dem/relion.git', preferred='true', tag='2.1')
+    version('2.0.3', git='https://github.com/3dem/relion.git', tag='2.0.3')
+    version('develop', git='https://github.com/3dem/relion.git')
+
 
     variant('gui', default=True, description="build the gui")
     variant('cuda', default=True, description="enable compute on gpu")
     variant('double', default=True, description="double precision (cpu) code")
     variant('double-gpu', default=False, description="double precision (gpu) code")
+    # added below cluster and desktop variants given some shared libraries in /usr/lib, like gpfs,
+    # are loaded by ld.so after compile which makes relion not work on desktops which may
+    # not be a member of gpfs cluster thus lack these given libraries
+    # https://github.com/spack/spack/issues/8384
+    variant('cluster', default=True, description="build relion for use in cluster")
+    variant('desktop', default=False, description="build relion for use in desktop")
```

Comment:
Are these mutually exclusive?

Reply:
Yes

Reply:
Then this should be a single multi-valued variant.

Reply:
Adding conflicts